js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,321 @@
+		return promise
+	}
+
+	// Obtain the underlying ArrayBuffer from the Uint8Array
+	ab, ok := intoObj.Get("buffer").Export().(goja.ArrayBuffer)
+	if !ok {
+		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		return promise
+	}
+
+	// To avoid concurrency linked to modifying the runtime's `into` buffer from multiple
+	// goroutines we make sure to work on a separate copy, and will copy the bytes back
+	// into the runtime's `into` buffer once the promise is resolved.
+	intoBytes := ab.Bytes()
+	buffer := make([]byte, len(intoBytes))
+
+	// We register a callback to be executed by the VU's runtime.
+	// This ensures that the modification of the JS runtime's `into` buffer
+	// occurs on the main thread, during the promise's resolution.
+	callback := f.vu.RegisterCallback()
+	go func() {
+		n, readErr := f.file.Read(buffer)
+		callback(func() error {
+			_ = copy(intoBytes[0:n], buffer)
+
+			// Read was successful, resolve early with the number of
+			// bytes read.
+			if readErr == nil {
+				resolve(n)
+				return nil
+			}
+
+			var fsErr *fsError
+			isFSErr := errors.As(readErr, &fsErr)
+
+			if !isFSErr {
+				reject(readErr)
+				return nil
+			}
+
+			if fsErr.kind == EOFError && n == 0 {
+				resolve(goja.Null())
+			} else {
+				resolve(n)
+			}
+
+			return nil
+		})
+	}()
+
+	return promise
+}
+
+// Seek seeks to the given `offset` in the file, under the given `whence` mode.
+//
+// The returned promise resolves to the new `offset` (position) within the file, which
+// is expressed in bytes from the selected start, current, or end position depending
+// the provided `whence`.
+func (f *File) Seek(offset goja.Value, whence goja.Value) *goja.Promise {
+	promise, resolve, reject := f.vu.Runtime().NewPromise()
+
+	intOffset, err := exportInt(f.vu.Runtime(), offset)
+	if err != nil {
+		reject(newFsError(TypeError, "seek() failed; reason: the offset argument "+err.Error()))
+		return promise
+	}
+
+	intWhence, err := exportInt(f.vu.Runtime(), whence)
+	if err != nil {
+		reject(newFsError(TypeError, "seek() failed; reason: the whence argument "+err.Error()))
+		return promise
+	}
+
+	seekMode := SeekMode(intWhence)
+	switch seekMode {
+	case SeekModeStart, SeekModeCurrent, SeekModeEnd:
+		// Valid modes, do nothing.
+	default:
+		reject(newFsError(TypeError, "seek() failed; reason: the whence argument must be a SeekMode"))
+		return promise
+	}
+
+	callback := f.vu.RegisterCallback()
+	go func() {
+		newOffset, err := f.file.Seek(intOffset, seekMode)
+		callback(func() error {
+			if err != nil {
+				reject(err)
+				return err
+			}
+
+			resolve(newOffset)
+			return nil
+		})
+	}()
+
+	return promise
+}
+
+func isUint8Array(rt *goja.Runtime, o *goja.Object) bool {
+	uint8ArrayConstructor := rt.Get("Uint8Array")
+	if isUint8Array := o.Get("constructor").SameAs(uint8ArrayConstructor); !isUint8Array {
+		return false
+	}
+
+	return true
+}
+
+//nolint:unparam
```

Comment:
I'm not getting why there aren't failing tests. :thinking:

The passed target `i` is not a pointer so the linter seems to be right here, you will end with always return `0`. Am I missing something?

Reply:
I have brought back the tests in https://github.com/grafana/k6/pull/3165/commits/d48fb673f8fa0bb4e4f8f36e8f1e2ae96373ab62

However I'm now back to my initial issue:
```
var i int64
if err := rt.ExportTo(v, &i); err != nil {
	return 0, errors.New("cannot be interpreted as integer")
}
```

Not returning an error when the function is called with 'abc' as a value, despite the goja documentation mentioning "Returns error if conversion is not possible.". Either a bug, or a surprising behavior (for me). Will look into it 👍🏻

Reply:
I've published a fix. I'll look into more details into the Goja behavior I've found surprising during cooldown 👍🏻